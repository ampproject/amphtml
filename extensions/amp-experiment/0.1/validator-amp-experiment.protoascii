#
# Copyright 2016 The AMP HTML Authors. All Rights Reserved.
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#      http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS-IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the license.
#

# Specific script tags for custom elements and runtime imports.
tags: {  # amp-experiment
  html_format: AMP
  tag_name: "SCRIPT"
  spec_name: "amp-experiment extension .js script"
  mandatory_parent: "HEAD"
  unique_warning: true
  extension_unused_unless_tag_present: "amp-experiment"
  attrs: {
    name: "async"
    mandatory: true
    value: ""
  }
  attrs: {
    name: "custom-element"
    mandatory: true
    value: "amp-experiment"
    dispatch_key: true
  }
  attrs: { name: "nonce" }
  attrs: {
    name: "src"
    mandatory: true
    value_regex: "https://cdn\\.ampproject\\.org/v0/amp-experiment-(latest|0\\.1).js"
  }
  attrs: {
    name: "type"
    value_casei: "text/javascript"
  }
  cdata: {
    blacklisted_cdata_regex: {
      regex: "."
      error_message: "contents"
    }
  }
  spec_url: "https://www.ampproject.org/docs/reference/extended/amp-experiment.html"
}
tags: {  # amp-experiment (json)
  html_format: AMP
  tag_name: "SCRIPT"
  spec_name: "amp-experiment extension .json script"
  mandatory_parent: "AMP-EXPERIMENT"
  attrs: { name: "nonce" }
  attrs: {
    name: "type"
    mandatory: true
    value_casei: "application/json"
    dispatch_key: true
  }
  cdata: {
    blacklisted_cdata_regex: {
      regex: "<!--"
      error_message: "html comments"
    }
  }
  spec_url: "https://www.ampproject.org/docs/reference/extended/amp-experiment.html"
}
tags: {  # <amp-experiment>
  html_format: AMP
  tag_name: "AMP-EXPERIMENT"
  disallowed_ancestor: "AMP-SIDEBAR"
  also_requires_tag: "amp-experiment extension .js script"
  unique: true
  spec_url: "https://www.ampproject.org/docs/reference/extended/amp-experiment.html"
}
