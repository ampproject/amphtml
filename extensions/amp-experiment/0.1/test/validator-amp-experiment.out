FAIL
|  <!--
|    Copyright 2016 The AMP HTML Authors. All Rights Reserved.
|
|    Licensed under the Apache License, Version 2.0 (the "License");
|    you may not use this file except in compliance with the License.
|    You may obtain a copy of the License at
|
|        http://www.apache.org/licenses/LICENSE-2.0
|
|    Unless required by applicable law or agreed to in writing, software
|    distributed under the License is distributed on an "AS-IS" BASIS,
|    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
|    See the License for the specific language governing permissions and
|    limitations under the license.
|  -->
|  <!--
|    Test Description:
|    This tests validation for the amp-experiment tag.
|  -->
|  <!doctype html>
|  <html âš¡>
|  <head>
|      <meta charset="utf-8">
|      <title>Experiment examples</title>
|      <link rel="canonical" href="amps.html">
|      <meta name="viewport" content="width=device-width,minimum-scale=1,initial-scale=1">
|      <link href="https://fonts.googleapis.com/css?family=Questrial" rel="stylesheet" type="text/css">
|      <script async custom-element="amp-experiment" src="https://cdn.ampproject.org/v0/amp-experiment-0.1.js"></script>
|       <style amp-boilerplate>body{-webkit-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-moz-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-ms-animation:-amp-start 8s steps(1,end) 0s 1 normal both;animation:-amp-start 8s steps(1,end) 0s 1 normal both}@-webkit-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-moz-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-ms-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-o-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}</style><noscript><style amp-boilerplate>body{-webkit-animation:none;-moz-animation:none;-ms-animation:none;animation:none}</style></noscript>
|      <script async src="https://cdn.ampproject.org/v0.js"></script>
|  </head>
|  <body>
|
|  <amp-experiment>
|      <script type="application/json">
|          {
|              "background-color-test": {
|                  "variants": {
|                      "yellow": 50,
|                      "green": 50
|                  }
|              }
|          }
|      </script>
|  </amp-experiment>
|
|  <amp-experiment>
>> ^~~~~~~~~
amp-experiment/0.1/test/validator-amp-experiment.html:47:0 The tag 'amp-experiment' appears more than once in the document. (see https://amp.dev/documentation/components/amp-experiment)
|      <script type="invalidtype"></script>
>>     ^~~~~~~~~
amp-experiment/0.1/test/validator-amp-experiment.html:48:4 Custom JavaScript is not allowed. (see https://amp.dev/documentation/guides-and-tutorials/learn/validation-workflow/validation_errors/#custom-javascript-is-not-allowed)
|  </amp-experiment>
|  </body>
|  </html>